Refactor delete analysis

henrikstranneheim opened the pull request:
## Description

Refactor delete analysis

### Added

- Tests

### Changed

- Refactor delete analysis and migrate to CRUD

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor_delete_analysis -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

diitaz93 commented on trailblazer/store/crud/delete.py:
```diff
@@ -10,3 +17,18 @@ def delete_analysis_jobs(self, analysis: Analysis) -> None:
         self.commit()
+
+    def delete_analysis(self, analysis_id: int, force: bool = False) -> None:
+        """Delete the analysis from the database. Also delete ongoing analysis if 'force' is True.
+        Raise: MissingAnalysis when no analysis and TrailblazerError for ongoing analysis for analysis id.
```
Would you consider formatting the error-raising description in docstrings like this? With the description of the exception in another line and preceded with a tab and the different exception types on different lines. We have not discussed this, so feel free to disregard it. If you like the suggestion, implement it in the text editor and not here, as GitHub does not allow me to add tabs, I just added some spaces
```suggestion
        """Delete the analysis from the database. Also delete ongoing analysis if 'force' is True.
        Raises: 
            MissingAnalysis when no analysis is found 
            TrailblazerError when the analysis is ongoing.
```

diitaz93 commented on trailblazer/store/crud/update.py:
```diff
@@ -92,7 +92,8 @@ def cancel_ongoing_analysis(
     ) -> None:
         """Cancel all ongoing slurm jobs associated with the analysis, and set analysis status to 'cancelled'.
-        Raise: TrailblazerError when no analysis or no ongoing analysis for analysis id."""
+        Raise: MissingAnalysis when no analysis and TrailblazerError for no ongoing analysis for analysis id.
```
same formatting suggestions applies here

henrikstranneheim replied:
Fixed

diitaz93 reviewed (approved):
Looks good 👍

henrikstranneheim commented:
### CLI and API ✅ 

```
θ62° [henrik.stranneheim@hasta:~] [S_base] 6s 130 $ trailblazer scan
Updated status justhusky - 278333: running
Status in Tower: hardygrouse - 278330 - 4zY4HUy9hew76p
Updated status hardygrouse - 278330: running
Error updating analysis for: case - civilsole : FileNotFoundError
Error updating analysis for: case - savingstag : FileNotFoundError
Error updating analysis for: case - HW2V2DRX2 : FileNotFoundError
Error updating analysis for: case - divinetomcat : PermissionError
Error updating analysis for: case - handyshrimp : FileNotFoundError
Error updating analysis for: case - setcolt : PermissionError
Error updating analysis for: case - proudfish : PermissionError
Error updating analysis for: case - funnyaphid : FileNotFoundError
Error updating analysis for: case - onestag : PermissionError
Error updating analysis for: case - newgopher : PermissionError
Error updating analysis for: case - helpedlab : MissingSqueueOutput
Error updating analysis for: case - goodbengal : MissingSqueueOutput
Error updating analysis for: case - grandaphid : PermissionError
All analyses updated!

```

<img width="1424" alt="Screenshot 2023-09-12 at 15 51 33" src="https://github.com/Clinical-Genomics/trailblazer/assets/584738/84770e47-1710-45d8-826b-e96e87233545">

```
62° [henrik.stranneheim@hasta:~] [S_base] 14s $ trailblazer delete  --cancel-jobs --force 278333
Cancelling job 5131550 - bwa_mem2
Cancelling job 5131549 - bwa_mem2
Cancelling job 5131548 - bwa_mem2
Cancelling job 5131551 - samtools_merge
Cancelling job 5131552 - samtools_merge
Cancelling job 5131553 - samtools_merge
Cancelling job 5131554 - markduplicates
Cancelling job 5131555 - markduplicates
Cancelling job 5131556 - markduplicates
Cancelling job 5131582 - cnvnator_ar
Cancelling job 5131583 - cnvnator_ar
Cancelling job 5131584 - cnvnator_ar
Cancelling job 5131586 - manta
Cancelling job 5131590 - sv_varianteffectpredictor
Cancelling job 5131591 - sv_vcfparser
Cancelling job 5131592 - sv_rankvariant
Cancelling job 5131605 - deepvariant
Cancelling job 5131606 - deepvariant
Cancelling job 5131607 - deepvariant
Cancelling job 5131608 - glnexus_merge
Cancelling job 5131624 - prepareforvariantannotationblock
Cancelling job 5131625 - rhocall_ar
Cancelling job 5131626 - bcftools_norm
Cancelling job 5131627 - variant_annotation
Cancelling job 5131637 - frequency_filter
Cancelling job 5131638 - cadd_ar
Cancelling job 5131639 - varianteffectpredictor
Cancelling job 5131640 - vcfparser_ar
Cancelling job 5131641 - rankvariant
Cancelling job 5131557 - chanjo_sexcheck
Cancelling job 5131558 - chanjo_sexcheck
Cancelling job 5131559 - chanjo_sexcheck
Cancelling job 5131560 - samtools_subsample_mt
Cancelling job 5131561 - samtools_subsample_mt
Cancelling job 5131562 - samtools_subsample_mt
Cancelling job 5131563 - tiddit_coverage
Cancelling job 5131564 - tiddit_coverage
Cancelling job 5131565 - tiddit_coverage
Cancelling job 5131566 - chromograph_cov
Cancelling job 5131567 - chromograph_cov
Cancelling job 5131568 - chromograph_cov
Cancelling job 5131569 - mitodel
Cancelling job 5131570 - mitodel
Cancelling job 5131571 - mitodel
Cancelling job 5131572 - sambamba_depth
Cancelling job 5131573 - sambamba_depth
Cancelling job 5131574 - sambamba_depth
Cancelling job 5131575 - smncopynumbercaller
Cancelling job 5131576 - picardtools_collectmultiplemetrics
Cancelling job 5131577 - picardtools_collectmultiplemetrics
Cancelling job 5131578 - picardtools_collectmultiplemetrics
Cancelling job 5131579 - picardtools_collecthsmetrics
Cancelling job 5131580 - picardtools_collecthsmetrics
Cancelling job 5131581 - picardtools_collecthsmetrics
Cancelling job 5131585 - expansionhunter
Cancelling job 5131587 - tiddit
Cancelling job 5131588 - sv_combinevariantcallsets
Cancelling job 5131589 - sv_annotate
Cancelling job 5131593 - sv_reformat
Cancelling job 5131594 - vcf2cytosure_ar
Cancelling job 5131595 - me_merge_bam
Cancelling job 5131596 - me_merge_bam
Cancelling job 5131597 - me_merge_bam
Cancelling job 5131598 - retroseq
Cancelling job 5131599 - retroseq
Cancelling job 5131600 - retroseq
Cancelling job 5131601 - me_merge_vcfs
Cancelling job 5131602 - me_annotate
Cancelling job 5131603 - me_varianteffectpredictor
Cancelling job 5131604 - me_filter
Cancelling job 5131609 - gatk_combinevariantcallsets
Cancelling job 5131610 - peddy_ar
Cancelling job 5131611 - plink
Cancelling job 5131612 - gatk_variantevalall
Cancelling job 5131613 - gatk_variantevalall
Cancelling job 5131614 - gatk_variantevalall
Cancelling job 5131615 - gatk_collectreadcounts
Cancelling job 5131616 - gatk_collectreadcounts
Cancelling job 5131617 - gatk_collectreadcounts
Cancelling job 5131618 - gatk_denoisereadcounts
Cancelling job 5131619 - gatk_denoisereadcounts
Cancelling job 5131620 - gatk_denoisereadcounts
Cancelling job 5131621 - gens_generatedata
Cancelling job 5131622 - gens_generatedata
Cancelling job 5131623 - gens_generatedata
Cancelling job 5131628 - rhocall_viz
Cancelling job 5131629 - rhocall_viz
Cancelling job 5131630 - rhocall_viz
Cancelling job 5131631 - chromograph_rhoviz
Cancelling job 5131632 - chromograph_rhoviz
Cancelling job 5131633 - chromograph_rhoviz
Cancelling job 5131634 - upd_ar
Cancelling job 5131635 - chromograph_upd
Cancelling job 5131636 - mt_annotation
Cancelling job 5131642 - endvariantannotationblock
Cancelling job 5131643 - gatk_variantevalexome
Cancelling job 5131644 - gatk_variantevalexome
Cancelling job 5131645 - gatk_variantevalexome
Cancelling job 5131646 - qccollect_ar
Cancelling job 5131647 - version_collect_ar
Cancelling job 5131648 - multiqc_ar
Cancelling job 5131649 - analysisrunstatus
Cancelling job 5131650 - sacct
Cancelling job 5131546 - fastqc_ar
Cancelling job 5131547 - fastqc_ar
Cancelling job 5131545 - fastqc_ar
Case justhusky - Analysis 278333: all ongoing jobs cancelled successfully!
Updated status justhusky - 278333: running
Deleting analysis: 278333, for case: justhusky
```

henrikstranneheim commented:
Deployed on Hasta:

```
Deployed in prod
```
